scalding-core/src/main/scala/com/twitter/scalding/typed/LookupJoin.scala

```diff
@@ -116,88 +118,44 @@ object LookupJoin extends Serializable {
   def withWindowRightSumming[T: Ordering, K: Ordering, V, JoinedV: Semigroup](left: TypedPipe[(T, (K, V))],
     right: TypedPipe[(T, (K, JoinedV))],
     reducers: Option[Int] = None)(gate: (T, T) => Boolean): TypedPipe[(T, (K, (V, Option[JoinedV])))] = {
-    /**
-     * Implicit ordering on an either that doesn't care about the
-     * actual container values, puts the lookups before the service writes
-     * Since we assume it takes non-zero time to do a lookup.
-     */
-    implicit def eitherOrd[T, U]: Ordering[Either[T, U]] =
-      new Ordering[Either[T, U]] {
-        def compare(l: Either[T, U], r: Either[T, U]) =
-          (l, r) match {
-            case (Left(_), Right(_)) => -1
-            case (Right(_), Left(_)) => 1
-            case (Left(_), Left(_)) => 0
-            case (Right(_), Right(_)) => 0
-          }
+    withWindowRightSummingAndOptionalPartition(
+      left, right, reducers)(gate, { _.cumulativeSum(reducers.getOrElse(-1)) })
+  }
+
+  def withWindowRightSummingAndPartition[T: Ordering, S: Ordering, K: Ordering, V, JoinedV: Semigroup](left: TypedPipe[(T, (K, V))],
+    right: TypedPipe[(T, (K, JoinedV))],
+    reducers: Option[Int] = None)(
+      gate: (T, T) => Boolean, partition: T => S): TypedPipe[(T, (K, (V, Option[JoinedV])))] = {
+    withWindowRightSummingAndOptionalPartition(
+      left, right, reducers)(gate, { _.cumulativeSum(reducers.getOrElse(-1), partition) })
+  }
+
+  private def withWindowRightSummingAndOptionalPartition[T: Ordering, K: Ordering, V, JoinedV: Semigroup](left: TypedPipe[(T, (K, V))],
+    right: TypedPipe[(T, (K, JoinedV))],
+    reducers: Option[Int] = None)(
+      gate: (T, T) => Boolean,
+      cumulate: (TypedPipe[(K, (T, (Last[Option[V]], Option[(Last[T], JoinedV)])))]) => TypedPipe[(K, (T, (Last[Option[V]], Option[(Last[T], JoinedV)])))]): TypedPipe[(T, (K, (V, Option[JoinedV])))] = {
+    implicit val gatedRightSum: Semigroup[(T, JoinedV)] = Semigroup.from {
```

Comment:
ah, you're right... however it would be if I introduced second T. A start T and an end T